src/SqlBinding/SqlAsyncCollector.cs

```diff
@@ -358,7 +358,7 @@ public async static Task<TableInformation> RetrieveTableInformationAsync(SqlConn
                     {
                         while (await rdr.ReadAsync())
                         {
-                            primaryKeys.Add(rdr[ColumnName].ToString());
+                            primaryKeys.Add(rdr[ColumnName].ToString().ToLowerInvariant());
```

Comment:
I don't quite understand the details of the bug, but by do we need to lowercase all the column names? Is there some comparison we're doing that doesn't allow us to use case-insensitivity?

Reply:
Our C# comparisons are case-insensitive, but some of the ones SQL is doing are not. 

Say your C# object data serializes into this JSON parameter, based on the casing of the POCO object:
`"[{"productid":5, "cost":0}]"`

but our T-SQL query looks like this, based on the casing of the SQL columns:
`WHEN MATCHED THEN UPDATE SET  ExistingData.ProductID = NewData.ProductID, ExistingData.Cost = NewData.Cost 
WHEN NOT MATCHED THEN INSERT (ProductID,Cost) VALUES (ProductID,Cost);`

the upsert fails, as SQL cannot find "ProductID" and "Cost" in the serialized JSON parameter

Reply:
Whereas if you have consistent casing in both the JSON parameter and the T-SQL merge command, SQL can properly match them, despite the potential mismatch between that casing ("productid") and the column name casing ("ProductID").

This may be a bug on their side, as the columns themselves are case-insensitive, but this was my way of getting around it for now

Reply:
I see. BTW, case-insensitivity in SQL actually depends on the collation of the database, so not all databases are case sensitive.